utils/autoprodtyper.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import argparse
+import json
+
+import ssg.build_yaml
+import ssg.products
+import ssg.rules
+import ssg.yaml
+import ssg.utils
+import ssg.rule_yaml
+
+from mod_prodtype import add_products
+
+SSG_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-j", "--json", type=str, action="store",
+                        default="build/rule_dirs.json", help="File to read "
+                        "json output of rule_dir_json from (defaults to "
+                        "build/rule_dirs.json")
+    parser.add_argument("-c", "--build-config-yaml", default="build/build_config.yml",
+                        help="YAML file with information about the build configuration. "
+                        "Defaults to build/build_config.yml")
+    parser.add_argument("-p", "--profiles-root",
+                        help="Override where to look for profile files.")
+    parser.add_argument("product", type=str, help="Product to add prod types for")
+    parser.add_argument("profile", type=str, help="Profile to iterate over")
+
+    return parser.parse_args()
+
+
+def autoprodtype(env_yaml, rule_dirs, profile_path, product):
+    profile = ssg.build_yaml.ProfileWithInlinePolicies.from_yaml(profile_path, env_yaml)
+
+    for rule_id in profile.selected + profile.unselected:
```

Comment:
```suggestion
    for rule_id in profile.selected + profile.unselected:
        print(rule_id)
```


It would be nice to print the rule ID out here. Or maybe in add_products.

 Because currently the output is somehow weird:

```
Current prodtype is empty, not adding the new prodtype.
Current prodtype is empty, not adding the new prodtype.
Current prodtype is empty, not adding the new prodtype.
Current prodtype is empty, not adding the new prodtype.
/home/jcerny/work/git/scap-security-guide/ssg/rule_yaml.py:132: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  return yaml.load(new_file)
Current prodtype: fedora,ol7,ol8,rhel7,rhel8,rhv4,sle12,sle15,wrlinux1019,rhcos4
New prodtype: fedora,ol7,ol8,rhcos4,rhel7,rhel8,rhel9,rhv4,sle12,sle15,wrlinux1019
```
It isn't clear to which rules the messages are related.

Reply:
For the following line:

> /home/jcerny/work/git/scap-security-guide/ssg/rule_yaml.py:132: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
>   return yaml.load(new_file)

Note that this PR actually depends on a change made in #6882. 

And yes, I should add output based on which rule was modified.

Reply:
I've updated the rule output to silence the empty prodtypes and make which rule was modified clearer.